core/include/keep.S

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014, STMicroelectronics International N.V.
```

Comment:
Well I'm not sure about this. I can see the benefit, but on the other hand doing so would violate a more general rule, which is that `.h` files define C constructs (function prototypes, macros...), while `.S` files define assembly constructs. So it's perfectly valid for an assembly file to include a `.h` file but only to benefit from C macros (for example: constants, UNWIND(x)). Assembly macros belong to `.S` files IMO.
@jenswi-linaro your opinion?